### PR TITLE
Usb hid mouse fix

### PR DESCRIPTION
**Pull request by rakons**
Fix USB hid-mouse example look and functionality.

### CODE REVIEW

**Review by jfischer-no — APPROVED** [review]

File: samples/subsys/usb/hid-mouse/src/main.c [hunk 1]
```diff
@@ -138,7 +138,8 @@ void main(void)
 
 	gpio_pin_configure(gpio, PIN,
 			   GPIO_DIR_IN | GPIO_INT |
-			   GPIO_INT_EDGE | GPIO_INT_DOUBLE_EDGE);
+			   GPIO_INT_EDGE | GPIO_INT_DOUBLE_EDGE |
+			   GPIO_PUD_PULL_UP);
```
Comment (aurel32) [inline samples/subsys/usb/hid-mouse/src/main.c]: If we enable the pull-up, it means that the default state is a `1` and and that pressing the button gives a `0`. I guess it's then possible to simplify the code by replacing `def_val` by `1`.

Comment (rakons) [reply samples/subsys/usb/hid-mouse/src/main.c]: I have used PIN_PUD referencing to SW0_GPIO_PIN_PUD instead.

**Comment by rakons** [comment]
@finikorg - can we go with this pull-request or should I change anything more there?

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/hid-mouse/src/main.c [hunk 2]
```diff
@@ -28,6 +28,15 @@
 #error SW0_GPIO_PIN needs to be set in board.h
 #endif
 
+/* The switch pin pull-up/down flags */
+#ifdef SW0_GPIO_PIN_PUD
+#define PIN_PUD SW0_GPIO_PIN_PUD
```
[the remark this answers is not in the document]
Comment (rakons) [reply samples/subsys/usb/hid-mouse/src/main.c]: Ok.. this is the reason that I saw some of the boards was using PUD and it goes from board.h and some FLAGS that comes from DTS. So we are waiting for #9551 then.

**Comment by finikorg** [comment]
@rakons looks OK to me, not tested yet, but lets wait for #9551.

File: samples/subsys/usb/hid-mouse/src/main.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (rakons) [reply samples/subsys/usb/hid-mouse/src/main.c]: @jfischer-phytec-iot - Could you approve after changes?

**Comment by rakons** [comment]
@jfischer-phytec-iot @finikorg @aurel32 Can anyone comment the question I have set above?

**Comment by aurel32** [comment]
recheck

**Comment by aurel32** [comment]
The test board recently got GPIO flags added. Let's see if it works.

**Review by aurel32 — APPROVED** [review]

**Comment by aurel32** [comment]
recheck

**Review by finikorg — APPROVED** [review]
LGTM, would be good to clean spaces/tabs in comments also.

**Comment by rakons** [comment]
@jfischer-phytec-iot Could you please check if the current solution is ok?  I have changed it two weeks above and it is still hanging here.

**Review by jfischer-no — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**